glustercli/cmd/snapshot-delete.go

```diff
@@ -19,24 +20,78 @@ var (
 		Use:   "delete <snapname>",
 		Short: snapshotDeleteHelpShort,
 		Long:  snapshotDeleteHelpLong,
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.RangeArgs(1, 3),
 		Run:   snapshotDeleteCmdRun,
 	}
 )
 
+var (
+	snapshotDeleteAllCmd = &cobra.Command{
+		Use:   "all [volname]",
+		Short: snapshotDeleteAllHelpShort,
+		Long:  snapshotDeleteAllHelpLong,
+		Args:  cobra.MaximumNArgs(1),
+		Run:   snapshotDeleteAllCmdRun,
+	}
+)
+
 func init() {
 	snapshotCmd.AddCommand(snapshotDeleteCmd)
+	snapshotDeleteCmd.AddCommand(snapshotDeleteAllCmd)
 }
 
-func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
-	snapname := args[0]
-
+func snapshotDelete(snapname string) error {
 	if err := client.SnapshotDelete(snapname); err != nil {
 		if GlobalFlag.Verbose {
 			log.WithError(err).WithField(
 				"snapshot", snapname).Error("snapshot delete failed")
 		}
-		failure("Snapshot delete failed", err, 1)
+		return err
 	}
 	fmt.Printf("%s Snapshot deleted successfully\n", snapname)
+	return nil
+}
+
+func snapshotDeleteCmdRun(cmd *cobra.Command, args []string) {
+
+	if len(args) != 1 {
+		//snapshotDeleteAllCmdRun will take care of else case.
+		return
+	}
+	snapname := args[0]
+	if err := snapshotDelete(snapname); err != nil {
+		failure("Snapshot delete failed", err, 1)
+	}
+}
+
+func snapshotDeleteAllCmdRun(cmd *cobra.Command, args []string) {
+	var snaps api.SnapListResp
+	var err error
+	volname := ""
+	if len(args) > 0 {
+		volname = args[0]
+	}
+
```

Comment:
@rafikc30 I think there is a need to throw an error if the volume is not provided.

Reply:
Volume is an optional argument. If omitted, then perform the deletion of all snapshots in the cluster. I have updated the example cli commands in commit message